torch/csrc/Module.cpp

```diff
@@ -57,6 +57,35 @@ static bool THPModule_loadClasses(PyObject *self)
 #undef ASSERT_NOT_NULL
 }
 
+static PyObject * THPModule_initNames(PyObject *self, PyObject *arg)
+{
+  static std::vector<std::string> names;
+
+  THPObjectPtr types = PySequence_Fast(arg, "expected a sequence");
+  if (!types) return NULL;
+
+  int num_classes = PySequence_Fast_GET_SIZE(types.get());
+  names.reserve(num_classes);
```

Comment:
I think it'd be better to use `names.size() + num_classes` and allow calling that function multiple times (that will be useful for distributed tensors).